device.go

```diff
@@ -57,8 +54,6 @@ var (
 // CCW variables
 var (
 	channelSubSystem = "/devices/css0"
-	sysCCWBusDir     = sysfsDir + "/bus/ccw/devices"
-	blkCCWSuffix     = "virtio"
```

Comment:
cc @alicefr @jschintag

Reply:
Hi @jschintag @alicefr 
I'm not familiar with the CCW bus devices represented in /sys and uevent, so for it's check, I need you guys input.

Reply:
@lifupan I need to understand a little you PR, up to now it's not really clear what you need. Do you have more specific question. I'm going also to apply your PRs and tested on s390x. Let you know

Reply:
Hi @alicefr,  I want to know how to match an "add" uevent for a specific "bus"  CCW block device.    For this patch,  only used the "channelSubSystem" to match with uevent's DevPath field, but this isn't specific to a specified "bus".

Reply:
So, here is the agent logs, just copy the part related to virtio-blk-ccw
```sh
Aug 15 15:25:29 s8360048 kata-proxy[59090]: time="2019-08-15T15:25:29.861930674+02:00" level=info msg="time=\"2019-08-15T13:25:29.849985844Z\" level=info msg=\"Received add uevent\" debug_console=false name=kata-agent pid=1 source=agent subsystem=udevlistener uevent-action=add uevent-devname= uevent-devpath=/devices/css0/0.0.0004 uevent-seqnum=999 uevent-subsystem=css\n" name=kata-proxy pid=59090 sandbox=f2cd85a60274e100c95aee63cc636cf2818bf3b7123b669c39d743da1c65bdeb source=agent
Aug 15 15:25:29 s8360048 kata-proxy[59090]: time="2019-08-15T15:25:29.863145785+02:00" level=info msg="time=\"2019-08-15T13:25:29.851268759Z\" level=info msg=\"Received add uevent\" debug_console=false name=kata-agent pid=1 source=agent subsystem=udevlistener uevent-action=add uevent-devname= uevent-devpath=/devices/css0/0.0.0004/0.0.0005 uevent-seqnum=1000 uevent-subsystem=ccw\n" name=kata-proxy pid=59090 sandbox=f2cd85a60274e100c95aee63cc636cf2818bf3b7123b669c39d743da1c65bdeb source=agent
Aug 15 15:25:29 s8360048 kata-proxy[59090]: time="2019-08-15T15:25:29.886704147+02:00" level=info msg="virtio_blk virtio4: [vda] 20971520 512-byte logical blocks (10.7 GB/10.0 GiB)\n" name=kata-proxy pid=59090 sandbox=f2cd85a60274e100c95aee63cc636cf2818bf3b7123b669c39d743da1c65bdeb source=agent
Aug 15 15:25:29 s8360048 kata-proxy[59090]: time="2019-08-15T15:25:29.891177246+02:00" level=info msg="time=\"2019-08-15T13:25:29.879229659Z\" level=info msg=\"Received add uevent\" debug_console=false name=kata-agent pid=1 source=agent subsystem=udevlistener uevent-action=add uevent-devname= uevent-devpath=/devices/css0/0.0.0004/0.0.0005/virtio4 uevent-seqnum=1002 uevent-subsystem=virtio\n" name=kata-proxy pid=59090 sandbox=f2cd85a60274e100c95aee63cc636cf2818bf3b7123b669c39d743da1c65bdeb source=agent
Aug 15 15:25:29 s8360048 kata-proxy[59090]: time="2019-08-15T15:25:29.894953075+02:00" level=info msg="time=\"2019-08-15T13:25:29.883063832Z\" level=info msg=\"Received add uevent\" debug_console=false name=kata-agent pid=1 source=agent subsystem=udevlistener uevent-action=add uevent-devname= uevent-devpath=\"/devices/virtual/bdi/254:0\" uevent-seqnum=1003 uevent-subsystem=bdi\n" name=kata-proxy pid=59090 sandbox=f2cd85a60274e100c95aee63cc636cf2818bf3b7123b669c39d743da1c65bdeb source=agent
Aug 15 15:25:29 s8360048 kata-proxy[59090]: time="2019-08-15T15:25:29.904203436+02:00" level=info msg="time=\"2019-08-15T13:25:29.892177609Z\" level=info msg=\"Received add uevent\" debug_console=false name=kata-agent pid=1 source=agent subsystem=udevlistener uevent-action=add uevent-devname=vda uevent-devpath=/devices/css0/0.0.0004/0.0.0005/virtio4/block/vda uevent-seqnum=1004 uevent-subsystem=block\n" name=kata-proxy pid=59090 sandbox=f2cd85a60274e100c95aee63cc636cf2818bf3b7123b669c39d743da1c65bdeb source=agent
```

Reply:
I thinks the simple way it's to try with your patch. Doing right now

Reply:
mmh it looks wrong, using your changes on s390x. I got:
```sh
docker run -ti --runtime kata-runtime alpine sh
docker: Error response from daemon: OCI runtime create failed: rpc error: code = DeadlineExceeded desc = Timeout reached after 3s waiting for device /devices/css0
```

Reply:
Hi @alicefr 
I made some changes, please have a try again, if it doesn't work, please paste the full log and let me debug.

Reply:
@lifupan it still doesn't work
docker:
```sh
docker run -ti --runtime kata-runtime alpine sh
docker: Error response from daemon: OCI runtime create failed: rpc error: code = DeadlineExceeded desc = Timeout reached after 3s waiting for device /devices/css0: unknown.
```
proxy:
```sh
Aug 16 09:02:21 s8360048 kata-proxy[25734]: time="2019-08-16T09:02:21.271618242+02:00" level=info msg=announce agent-logs-socket=/run/vc/vm/7f1a1f9e64edbbe857d6883a0d013f58800cbdf832c4b7c9a7a51ef6e2707f2f/console.sock channel-mux-socket=/run/vc/vm/7f1a1f9e64edbbe857d6883a0d013f58800cbdf832c4b7c9a7a51ef6e2707f2f/kata.sock debug=true log-level=debug name=kata-proxy pid=25734 proxy-listen-socket="unix:///run/vc/sbs/7f1a1f9e64edbbe857d6883a0d013f58800cbdf832c4b7c9a7a51ef6e2707f2f/proxy.sock" sandbox=7f1a1f9e64edbbe857d6883a0d013f58800cbdf832c4b7c9a7a51ef6e2707f2f source=proxy version=1.7.0-alpha1-ce8ee7147c699521d962469377d93d338b79f22e
Aug 16 09:02:27 s8360048 kata-proxy[25734]: time="2019-08-16T09:02:27.56052067+02:00" level=info msg="random: crng init done\n" name=kata-proxy pid=25734 sandbox=7f1a1f9e64edbbe857d6883a0d013f58800cbdf832c4b7c9a7a51ef6e2707f2f source=agent
Aug 16 09:02:27 s8360048 kata-proxy[25734]: time="2019-08-16T09:02:27.570596424+02:00" level=info msg="registered taskstats version 1\n" name=kata-proxy pid=25734 sandbox=7f1a1f9e64edbbe857d6883a0d013f58800cbdf832c4b7c9a7a51ef6e2707f2f source=agent
Aug 16 09:02:27 s8360048 kata-proxy[25734]: time="2019-08-16T09:02:27.575609729+02:00" level=info msg="Btrfs loaded, crc32c=crc32c-vx\n" name=kata-proxy pid=25734 sandbox=7f1a1f9e64edbbe857d6883a0d013f58800cbdf832c4b7c9a7a51ef6e2707f2f source=agent
Aug 16 09:02:27 s8360048 kata-proxy[25734]: time="2019-08-16T09:02:27.582599792+02:00" level=info msg="Freeing unused kernel memory: 2560K\n" name=kata-proxy pid=25734 sandbox=7f1a1f9e64edbbe857d6883a0d013f58800cbdf832c4b7c9a7a51ef6e2707f2f source=agent
Aug 16 09:02:27 s8360048 kata-proxy[25734]: time="2019-08-16T09:02:27.595347186+02:00" level=info msg="Write protected read-only-after-init data: 20k\n" name=kata-proxy pid=25734 sandbox=7f1a1f9e64edbbe857d6883a0d013f58800cbdf832c4b7c9a7a51ef6e2707f2f source=agent
Aug 16 09:02:27 s8360048 kata-proxy[25734]: time="2019-08-16T09:02:27.595863291+02:00" level=info msg="Run /init as init process\n" name=kata-proxy pid=25734 sandbox=7f1a1f9e64edbbe857d6883a0d013f58800cbdf832c4b7c9a7a51ef6e2707f2f source=agent
Aug 16 09:02:27 s8360048 kata-proxy[25734]: time="2019-08-16T09:02:27.684007478+02:00" level=info msg="announce: name=\"kata-agent\",pid=\"1\",source=\"agent\",version=\"1.9.0-alpha0-35a9cac6bcfc07b856101d683da59201ea9152d6\",device-handlers=\"nvdimm,mmioblk,blk,blk-ccw,scsi\",storage-handlers=\"mmioblk,scsi,ephemeral,local,9p,virtio-fs,blk,blk-ccw\",system-memory=\"2029028 kB\"\n" name=kata-proxy pid=25734 sandbox=7f1a1f9e64edbbe857d6883a0d013f58800cbdf832c4b7c9a7a51ef6e2707f2f source=agent
Aug 16 09:02:27 s8360048 kata-proxy[25734]: time="2019-08-16T09:02:27.689816479+02:00" level=info msg="announce: name=\"kata-agent\",pid=\"1\",source=\"agent\",version=\"1.9.0-alpha0-35a9cac6bcfc07b856101d683da59201ea9152d6\",device-handlers=\"blk,blk-ccw,scsi,nvdimm,mmioblk\",storage-handlers=\"virtio-fs,blk,blk-ccw,mmioblk,scsi,ephemeral,local,9p\",system-memory=\"2029028 kB\"\n" name=kata-proxy pid=25734 sandbox=7f1a1f9e64edbbe857d6883a0d013f58800cbdf832c4b7c9a7a51ef6e2707f2f source=agent
Aug 16 09:02:27 s8360048 kata-proxy[25734]: time="2019-08-16T09:02:27.709229854+02:00" level=info msg="time=\"2019-08-16T07:02:27.706162843Z\" level=debug msg=\"Serial channel type detected\" debug_console=false name=kata-agent pid=1 source=agent\n" name=kata-proxy pid=25734 sandbox=7f1a1f9e64edbbe857d6883a0d013f58800cbdf832c4b7c9a7a51ef6e2707f2f source=agent
Aug 16 09:02:27 s8360048 kata-proxy[25734]: time="2019-08-16T09:02:27.712174316+02:00" level=info msg="time=\"2019-08-16T07:02:27.711136358Z\" level=info msg=\"Started listening for uevents\" debug_console=false name=kata-agent pid=1 source=agent subsystem=udevlistener\n" name=kata-proxy pid=25734 sandbox=7f1a1f9e64edbbe857d6883a0d013f58800cbdf832c4b7c9a7a51ef6e2707f2f source=agent
Aug 16 09:02:27 s8360048 kata-proxy[25734]: time="2019-08-16T09:02:27.713979667+02:00" level=info msg="time=\"2019-08-16T07:02:27.713023947Z\" level=info msg=\"agent grpc server starts\" debug_console=false name=kata-agent pid=1 source=agent\n" name=kata-proxy pid=25734 sandbox=7f1a1f9e64edbbe857d6883a0d013f58800cbdf832c4b7c9a7a51ef6e2707f2f source=agent
Aug 16 09:02:27 s8360048 kata-proxy[25734]: time="2019-08-16T09:02:27.71615623+02:00" level=info msg="time=\"2019-08-16T07:02:27.715326605Z\" level=debug msg=\"New serial channel event\" debug_console=false events=4 name=kata-agent pid=1 source=agent\n" name=kata-proxy pid=25734 sandbox=7f1a1f9e64edbbe857d6883a0d013f58800cbdf832c4b7c9a7a51ef6e2707f2f source=agent
Aug 16 09:02:27 s8360048 kata-proxy[25734]: time="2019-08-16T09:02:27.718266792+02:00" level=info msg="time=\"2019-08-16T07:02:27.717312069Z\" level=info msg=\"Waiting for stopServer signal...\" debug_console=false name=kata-agent pid=1 source=agent subsystem=stopserverwatcher\n" name=kata-proxy pid=25734 sandbox=7f1a1f9e64edbbe857d6883a0d013f58800cbdf832c4b7c9a7a51ef6e2707f2f source=agent
Aug 16 09:02:27 s8360048 kata-proxy[25734]: time="2019-08-16T09:02:27.723915228+02:00" level=info msg="time=\"2019-08-16T07:02:27.723051732Z\" level=debug msg=\"new request\" debug_console=false name=kata-agent pid=1 req= request=/grpc.Health/Check source=agent\n" name=kata-proxy pid=25734 sandbox=7f1a1f9e64edbbe857d6883a0d013f58800cbdf832c4b7c9a7a51ef6e2707f2f source=agent
Aug 16 09:02:27 s8360048 kata-proxy[25734]: time="2019-08-16T09:02:27.72548798+02:00" level=info msg="time=\"2019-08-16T07:02:27.724236256Z\" level=debug msg=\"request end\" debug_console=false duration=\"61.41µs\" name=kata-agent pid=1 request=/grpc.Health/Check resp=\"status:SERVING \" source=agent\n" name=kata-proxy pid=25734 sandbox=7f1a1f9e64edbbe857d6883a0d013f58800cbdf832c4b7c9a7a51ef6e2707f2f source=agent
Aug 16 09:02:27 s8360048 kata-proxy[25734]: time="2019-08-16T09:02:27.763875246+02:00" level=info msg="time=\"2019-08-16T07:02:27.748437694Z\" level=debug msg=\"new request\" debug_console=false name=kata-agent pid=1 req=\"interface:<device:\\\"eth0\\\" name:\\\"eth0\\\" IPAddresses:<address:\\\"172.17.0.2\\\" mask:\\\"16\\\" > mtu:1500 hwAddr:\\\"02:42:ac:11:00:02\\\" > \" request=/grpc.AgentService/UpdateInterface source=agent\n" name=kata-proxy pid=25734 sandbox=7f1a1f9e64edbbe857d6883a0d013f58800cbdf832c4b7c9a7a51ef6e2707f2f source=agent
Aug 16 09:02:27 s8360048 kata-proxy[25734]: time="2019-08-16T09:02:27.790080747+02:00" level=info msg="time=\"2019-08-16T07:02:27.765010567Z\" level=info msg=\"Getting interface from MAC address\" debug_console=false interface-name=eth0 mac-address=\"02:42:ac:11:00:02\" name=kata-agent pci-address= pid=1 source=agent\n" name=kata-proxy pid=25734 sandbox=7f1a1f9e64edbbe857d6883a0d013f58800cbdf832c4b7c9a7a51ef6e2707f2f source=agent
Aug 16 09:02:27 s8360048 kata-proxy[25734]: time="2019-08-16T09:02:27.796372168+02:00" level=info msg="time=\"2019-08-16T07:02:27.788950171Z\" level=info msg=\"Link found\" debug_console=false interface-name=eth0 link=\"&{LinkAttrs:{Index:3 MTU:1500 TxQLen:1000 Name:eth0 HardwareAddr:02:42:ac:11:00:02 Flags:broadcast|multicast RawFlags:4098 ParentIndex:0 MasterIndex:0 Namespace:<nil> Alias: Statistics:0xc000218100 Promisc:0 Xdp:0xc000089ac0 EncapType:ether Protinfo:<nil> OperState:down NetNsID:0 NumTxQueues:0 NumRxQueues:0 Vfs:[]}}\" mac-address=\"02:42:ac:11:00:02\" name=kata-agent pci-address= pid=1 source=agent\n" name=kata-proxy pid=25734 sandbox=7f1a1f9e64edbbe857d6883a0d013f58800cbdf832c4b7c9a7a51ef6e2707f2f source=agent
Aug 16 09:02:27 s8360048 kata-proxy[25734]: time="2019-08-16T09:02:27.856683877+02:00" level=info msg="IPv6: ADDRCONF(NETDEV_UP): eth0: link is not ready\n" name=kata-proxy pid=25734 sandbox=7f1a1f9e64edbbe857d6883a0d013f58800cbdf832c4b7c9a7a51ef6e2707f2f source=agent
Aug 16 09:02:27 s8360048 kata-proxy[25734]: time="2019-08-16T09:02:27.856733993+02:00" level=info msg="8021q: adding VLAN 0 to HW filter on device eth0\n" name=kata-proxy pid=25734 sandbox=7f1a1f9e64edbbe857d6883a0d013f58800cbdf832c4b7c9a7a51ef6e2707f2f source=agent
Aug 16 09:02:27 s8360048 kata-proxy[25734]: time="2019-08-16T09:02:27.867060545+02:00" level=info msg="time=\"2019-08-16T07:02:27.860479455Z\" level=debug msg=\"request end\" debug_console=false duration=95.342375ms name=kata-agent pid=1 request=/grpc.AgentService/UpdateInterface resp=\"device:\\\"eth0\\\" name:\\\"eth0\\\" IPAddresses:<address:\\\"172.17.0.2\\\" mask:\\\"16\\\" > mtu:1500 hwAddr:\\\"02:42:ac:11:00:02\\\" \" source=agent\n" name=kata-proxy pid=25734 sandbox=7f1a1f9e64edbbe857d6883a0d013f58800cbdf832c4b7c9a7a51ef6e2707f2f source=agent
Aug 16 09:02:27 s8360048 kata-proxy[25734]: time="2019-08-16T09:02:27.881588793+02:00" level=info msg="time=\"2019-08-16T07:02:27.880761129Z\" level=debug msg=\"new request\" debug_console=false name=kata-agent pid=1 req=\"routes:<Routes:<gateway:\\\"172.17.0.1\\\" device:\\\"eth0\\\" > > \" request=/grpc.AgentService/UpdateRoutes source=agent\n" name=kata-proxy pid=25734 sandbox=7f1a1f9e64edbbe857d6883a0d013f58800cbdf832c4b7c9a7a51ef6e2707f2f source=agent
Aug 16 09:02:27 s8360048 kata-proxy[25734]: time="2019-08-16T09:02:27.894832607+02:00" level=info msg="time=\"2019-08-16T07:02:27.893922176Z\" level=debug msg=\"request end\" debug_console=false duration=11.890471ms name=kata-agent pid=1 request=/grpc.AgentService/UpdateRoutes resp=\"Routes:<gateway:\\\"172.17.0.1\\\" device:\\\"eth0\\\" > \" source=agent\n" name=kata-proxy pid=25734 sandbox=7f1a1f9e64edbbe857d6883a0d013f58800cbdf832c4b7c9a7a51ef6e2707f2f source=agent
Aug 16 09:02:27 s8360048 kata-proxy[25734]: time="2019-08-16T09:02:27.989923979+02:00" level=info msg="time=\"2019-08-16T07:02:27.989105842Z\" level=debug msg=\"new request\" debug_console=false name=kata-agent pid=1 req=\"hostname:\\\"7f1a1f9e64ed\\\" storages:<driver:\\\"9p\\\" source:\\\"kataShared\\\" fstype:\\\"9p\\\" options:\\\"trans=virtio,version=9p2000.L,cache=mmap\\\" options:\\\"nodev\\\" options:\\\"msize=8192\\\" mount_point:\\\"/run/kata-containers/shared/containers/\\\" > storages:<driver:\\\"ephemeral\\\" source:\\\"shm\\\" fstype:\\\"tmpfs\\\" options:\\\"noexec\\\" options:\\\"nosuid\\\" options:\\\"nodev\\\" options:\\\"mode=1777\\\" options:\\\"size=67108864\\\" mount_point:\\\"/run/kata-containers/sandbox/shm\\\" > sandbox_id:\\\"7f1a1f9e64edbbe857d6883a0d013f58800cbdf832c4b7c9a7a51ef6e2707f2f\\\" \" request=/grpc.AgentService/CreateSandbox source=agent\n" name=kata-proxy pid=25734 sandbox=7f1a1f9e64edbbe857d6883a0d013f58800cbdf832c4b7c9a7a51ef6e2707f2f source=agent
Aug 16 09:02:28 s8360048 kata-proxy[25734]: time="2019-08-16T09:02:28.000667177+02:00" level=info msg="time=\"2019-08-16T07:02:27.999775197Z\" level=debug debug_console=false mount-destination=/run/kata-containers/shared/containers/ mount-flags=4 mount-fstype=9p mount-options=\"trans=virtio,version=9p2000.L,cache=mmap,msize=8192\" mount-source=kataShared name=kata-agent pid=1 sandbox=7f1a1f9e64edbbe857d6883a0d013f58800cbdf832c4b7c9a7a51ef6e2707f2f source=agent\n" name=kata-proxy pid=25734 sandbox=7f1a1f9e64edbbe857d6883a0d013f58800cbdf832c4b7c9a7a51ef6e2707f2f source=agent
Aug 16 09:02:28 s8360048 kata-proxy[25734]: time="2019-08-16T09:02:28.009957133+02:00" level=info msg="time=\"2019-08-16T07:02:28.009045839Z\" level=info msg=\"Received add uevent\" debug_console=false name=kata-agent pid=1 source=agent subsystem=udevlistener uevent-action=add uevent-devname= uevent-devpath=/devices/virtual/bdi/9p-1 uevent-seqnum=996 uevent-subsystem=bdi\n" name=kata-proxy pid=25734 sandbox=7f1a1f9e64edbbe857d6883a0d013f58800cbdf832c4b7c9a7a51ef6e2707f2f source=agent
Aug 16 09:02:28 s8360048 kata-proxy[25734]: time="2019-08-16T09:02:28.012020541+02:00" level=info msg="time=\"2019-08-16T07:02:28.011043397Z\" level=debug debug_console=false mount-destination=/run/kata-containers/sandbox/shm mount-flags=14 mount-fstype=tmpfs mount-options=\"mode=1777,size=67108864\" mount-source=shm name=kata-agent pid=1 sandbox=7f1a1f9e64edbbe857d6883a0d013f58800cbdf832c4b7c9a7a51ef6e2707f2f source=agent\n" name=kata-proxy pid=25734 sandbox=7f1a1f9e64edbbe857d6883a0d013f58800cbdf832c4b7c9a7a51ef6e2707f2f source=agent
Aug 16 09:02:28 s8360048 kata-proxy[25734]: time="2019-08-16T09:02:28.050352854+02:00" level=info msg="time=\"2019-08-16T07:02:28.012703145Z\" level=debug msg=\"request end\" debug_console=false duration=22.973267ms name=kata-agent pid=1 request=/grpc.AgentService/CreateSandbox resp=\"&Empty{XXX_unrecognized:[],}\" sandbox=7f1a1f9e64edbbe857d6883a0d013f58800cbdf832c4b7c9a7a51ef6e2707f2f source=agent\n" name=kata-proxy pid=25734 sandbox=7f1a1f9e64edbbe857d6883a0d013f58800cbdf832c4b7c9a7a51ef6e2707f2f source=agent
Aug 16 09:02:28 s8360048 kata-proxy[25734]: time="2019-08-16T09:02:28.092547355+02:00" level=info msg="time=\"2019-08-16T07:02:28.091657084Z\" level=debug msg=\"new request\" debug_console=false name=kata-agent pid=1 req=\"mem_block_size:true mem_hotplug_probe:true \" request=/grpc.AgentService/GetGuestDetails sandbox=7f1a1f9e64edbbe857d6883a0d013f58800cbdf832c4b7c9a7a51ef6e2707f2f source=agent\n" name=kata-proxy pid=25734 sandbox=7f1a1f9e64edbbe857d6883a0d013f58800cbdf832c4b7c9a7a51ef6e2707f2f source=agent
Aug 16 09:02:28 s8360048 kata-proxy[25734]: time="2019-08-16T09:02:28.096846745+02:00" level=info msg="time=\"2019-08-16T07:02:28.095905897Z\" level=debug msg=\"request end\" debug_console=false duration=1.974312ms name=kata-agent pid=1 request=/grpc.AgentService/GetGuestDetails resp=\"mem_block_size_bytes:268435456 agent_details:<version:\\\"1.9.0-alpha0-35a9cac6bcfc07b856101d683da59201ea9152d6\\\" init_daemon:true device_handlers:\\\"mmioblk\\\" device_handlers:\\\"blk\\\" device_handlers:\\\"blk-ccw\\\" device_handlers:\\\"scsi\\\" device_handlers:\\\"nvdimm\\\" storage_handlers:\\\"9p\\\" storage_handlers:\\\"virtio-fs\\\" storage_handlers:\\\"blk\\\" storage_handlers:\\\"blk-ccw\\\" storage_handlers:\\\"mmioblk\\\" storage_handlers:\\\"scsi\\\" storage_handlers:\\\"ephemeral\\\" storage_handlers:\\\"local\\\" > \" sandbox=7f1a1f9e64edbbe857d6883a0d013f58800cbdf832c4b7c9a7a51ef6e2707f2f source=agent\n" name=kata-proxy pid=25734 sandbox=7f1a1f9e64edbbe857d6883a0d013f58800cbdf832c4b7c9a7a51ef6e2707f2f source=agent
Aug 16 09:02:28 s8360048 kata-proxy[25734]: time="2019-08-16T09:02:28.12545448+02:00" level=info msg="time=\"2019-08-16T07:02:28.118087489Z\" level=debug msg=\"new request\" debug_console=false name=kata-agent pid=1 req= request=/grpc.AgentService/OnlineCPUMem sandbox=7f1a1f9e64edbbe857d6883a0d013f58800cbdf832c4b7c9a7a51ef6e2707f2f source=agent\n" name=kata-proxy pid=25734 sandbox=7f1a1f9e64edbbe857d6883a0d013f58800cbdf832c4b7c9a7a51ef6e2707f2f source=agent
Aug 16 09:02:28 s8360048 kata-proxy[25734]: time="2019-08-16T09:02:28.132120541+02:00" level=info msg="time=\"2019-08-16T07:02:28.12699551Z\" level=debug msg=\"request end\" debug_console=false duration=\"73.09µs\" name=kata-agent pid=1 request=/grpc.AgentService/OnlineCPUMem resp=\"&Empty{XXX_unrecognized:[],}\" sandbox=7f1a1f9e64edbbe857d6883a0d013f58800cbdf832c4b7c9a7a51ef6e2707f2f source=agent\n" name=kata-proxy pid=25734 sandbox=7f1a1f9e64edbbe857d6883a0d013f58800cbdf832c4b7c9a7a51ef6e2707f2f source=agent
Aug 16 09:02:28 s8360048 kata-proxy[25734]: time="2019-08-16T09:02:28.326222687+02:00" level=info msg="Built 1 zonelists, mobility grouping on.  Total pages: 568569\n" name=kata-proxy pid=25734 sandbox=7f1a1f9e64edbbe857d6883a0d013f58800cbdf832c4b7c9a7a51ef6e2707f2f source=agent
Aug 16 09:02:28 s8360048 kata-proxy[25734]: time="2019-08-16T09:02:28.346283604+02:00" level=info msg="Policy zone: Normal\n" name=kata-proxy pid=25734 sandbox=7f1a1f9e64edbbe857d6883a0d013f58800cbdf832c4b7c9a7a51ef6e2707f2f source=agent
Aug 16 09:02:28 s8360048 kata-proxy[25734]: time="2019-08-16T09:02:28.349754293+02:00" level=info msg="crw_info : CRW reports slct=0, oflw=0, chn=1, rsc=3, anc=0, erc=4, rsid=4\n" name=kata-proxy pid=25734 sandbox=7f1a1f9e64edbbe857d6883a0d013f58800cbdf832c4b7c9a7a51ef6e2707f2f source=agent
Aug 16 09:02:28 s8360048 kata-proxy[25734]: time="2019-08-16T09:02:28.35043289+02:00" level=info msg="crw_info : CRW reports slct=0, oflw=0, chn=0, rsc=3, anc=0, erc=4, rsid=0\n" name=kata-proxy pid=25734 sandbox=7f1a1f9e64edbbe857d6883a0d013f58800cbdf832c4b7c9a7a51ef6e2707f2f source=agent
Aug 16 09:02:28 s8360048 kata-proxy[25734]: time="2019-08-16T09:02:28.451118542+02:00" level=info msg="time=\"2019-08-16T07:02:28.449836317Z\" level=debug msg=\"new request\" debug_console=false name=kata-agent pid=1 req=\"container_id:\\\"7f1a1f9e64edbbe857d6883a0d013f58800cbdf832c4b7c9a7a51ef6e2707f2f\\\" exec_id:\\\"7f1a1f9e64edbbe857d6883a0d013f58800cbdf832c4b7c9a7a51ef6e2707f2f\\\" storages:<driver:\\\"blk-ccw\\\" source:\\\"0.0.0005\\\" fstype:\\\"xfs\\\" options:\\\"nouuid\\\" mount_point:\\\"/run/kata-containers/shared/containers/7f1a1f9e64edbbe857d6883a0d013f58800cbdf832c4b7c9a7a51ef6e2707f2f\\\" > OCI:<Version:\\\"1.0.1\\\" Process:<Terminal:true User:<AdditionalGids:0 AdditionalGids:1 AdditionalGids:2 AdditionalGids:3 AdditionalGids:4 AdditionalGids:6 AdditionalGids:10 AdditionalGids:11 AdditionalGids:20 AdditionalGids:26 AdditionalGids:27 > Args:\\\"sh\\\" Env:\\\"PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin\\\" Env:\\\"HOSTNAME=7f1a1f9e64ed\\\" Env:\\\"TERM=xterm\\\" Cwd:\\\"/\\\" Capabilities:<Bounding:\\\"CAP_CHOWN\\\" Bounding:\\\"CAP_DAC_OVERRIDE\\\" Bounding:\\\"CAP_FSETID\\\" Bounding:\\\"CAP_FOWNER\\\" Bounding:\\\"CAP_MKNOD\\\" Bounding:\\\"CAP_NET_RAW\\\" Bounding:\\\"CAP_SETGID\\\" Bounding:\\\"CAP_SETUID\\\" Bounding:\\\"CAP_SETFCAP\\\" Bounding:\\\"CAP_SETPCAP\\\" Bounding:\\\"CAP_NET_BIND_SERVICE\\\" Bounding:\\\"CAP_SYS_CHROOT\\\" Bounding:\\\"CAP_KILL\\\" Bounding:\\\"CAP_AUDIT_WRITE\\\" Effective:\\\"CAP_CHOWN\\\" Effective:\\\"CAP_DAC_OVERRIDE\\\" Effective:\\\"CAP_FSETID\\\" Effective:\\\"CAP_FOWNER\\\" Effective:\\\"CAP_MKNOD\\\" Effective:\\\"CAP_NET_RAW\\\" Effective:\\\"CAP_SETGID\\\" Effective:\\\"CAP_SETUID\\\" Effective:\\\"CAP_SETFCAP\\\" Effective:\\\"CAP_SETPCAP\\\" Effective:\\\"CAP_NET_BIND_SERVICE\\\" Effective:\\\"CAP_SYS_CHROOT\\\" Effective:\\\"CAP_KILL\\\" Effective:\\\"CAP_AUDIT_WRITE\\\" Inheritable:\\\"CAP_CHOWN\\\" Inheritable:\\\"CAP_DAC_OVERRIDE\\\" Inheritable:\\\"CAP_FSETID\\\" Inheritable:\\\"CAP_FOWNER\\\" Inheritable:\\\"CAP_MKNOD\\\" Inheritable:\\\"CAP_NET_RAW\\\" Inheritable:\\\"CAP_SETGID\\\" Inheritable:\\\"CAP_SETUID\\\" Inheritable:\\\"CAP_SETFCAP\\\" Inheritable:\\\"CAP_SETPCAP\\\" Inheritable:\\\"CAP_NET_BIND_SERVICE\\\" Inheritable:\\\"CAP_SYS_CHROOT\\\" Inheritable:\\\"CAP_KILL\\\" Inheritable:\\\"CAP_AUDIT_WRITE\\\" Permitted:\\\"CAP_CHOWN\\\" Permitted:\\\"CAP_DAC_OVERRIDE\\\" Permitted:\\\"CAP_FSETID\\\" Permitted:\\\"CAP_FOWNER\\\" Permitted:\\\"CAP_MKNOD\\\" Permitted:\\\"CAP_NET_RAW\\\" Permitted:\\\"CAP_SETGID\\\" Permitted:\\\"CAP_SETUID\\\" Permitted:\\\"CAP_SETFCAP\\\" Permitted:\\\"CAP_SETPCAP\\\" Permitted:\\\"CAP_NET_BIND_SERVICE\\\" Permitted:\\\"CAP_SYS_CHROOT\\\" Permitted:\\\"CAP_KILL\\\" Permitted:\\\"CAP_AUDIT_WRITE\\\" > ApparmorProfile:\\\"docker-default\\\" > Root:<Path:\\\"/run/kata-containers/shared/containers/7f1a1f9e64edbbe857d6883a0d013f58800cbdf832c4b7c9a7a51ef6e2707f2f/rootfs\\\" > Hostname:\\\"7f1a1f9e64ed\\\" Mounts:<destination:\\\"/proc\\\" source:\\\"proc\\\" type:\\\"proc\\\" options:\\\"nosuid\\\" options:\\\"noexec\\\" options:\\\"nodev\\\" > Mounts:<destination:\\\"/dev\\\" source:\\\"tmpfs\\\" type:\\\"tmpfs\\\" options:\\\"nosuid\\\" options:\\\"strictatime\\\" options:\\\"mode=755\\\" options:\\\"size=65536k\\\" > Mounts:<destination:\\\"/dev/pts\\\" source:\\\"devpts\\\" type:\\\"devpts\\\" options:\\\"nosuid\\\" options:\\\"noexec\\\" options:\\\"newinstance\\\" options:\\\"ptmxmode=0666\\\" options:\\\"mode=0620\\\" options:\\\"gid=5\\\" > Mounts:<destination:\\\"/sys\\\" source:\\\"sysfs\\\" type:\\\"sysfs\\\" options:\\\"nosuid\\\" options:\\\"noexec\\\" options:\\\"nodev\\\" options:\\\"ro\\\" > Mounts:<destination:\\\"/sys/fs/cgroup\\\" source:\\\"cgroup\\\" type:\\\"cgroup\\\" options:\\\"ro\\\" options:\\\"nosuid\\\" options:\\\"noexec\\\" options:\\\"nodev\\\" > Mounts:<destination:\\\"/dev/mqueue\\\" source:\\\"mqueue\\\" type:\\\"mqueue\\\" options:\\\"nosuid\\\" options:\\\"noexec\\\" options:\\\"nodev\\\" > Mounts:<destination:\\\"/etc/resolv.conf\\\" source:\\\"/run/kata-containers/shared/containers/7f1a1f9e64edbbe857d6883a0d013f58800cbdf832c4b7c9a7a51ef6e2707f2f-a9ff7da281c5332e-resolv.conf\\\" type:\\\"bind\\\" options:\\\"rbind\\\" options:\\\"rprivate\\\" > Mounts:<destination:\\\"/etc/hostname\\\" source:\\\"/run/kata-containers/shared/containers/7f1a1f9e64edbbe857d6883a0d013f58800cbdf832c4b7c9a7a51ef6e2707f2f-f76469277235d0cb-hostname\\\" type:\\\"bind\\\" options:\\\"rbind\\\" options:\\\"rprivate\\\" > Mounts:<destination:\\\"/etc/hosts\\\" source:\\\"/run/kata-containers/shared/containers/7f1a1f9e64edbbe857d6883a0d013f58800cbdf832c4b7c9a7a51ef6e2707f2f-64180973eb4f0d8d-hosts\\\" type:\\\"bind\\\" options:\\\"rbind\\\" options:\\\"rprivate\\\" > Mounts:<destination:\\\"/dev/shm\\\" source:\\\"/run/kata-containers/sandbox/shm\\\" type:\\\"bind\\\" options:\\\"rbind\\\" > Linux:<Resources:<Memory:<> CPU:<> > CgroupsPath:\\\"/docker/7f1a1f9e64edbbe857d6883a0d013f58800cbdf832c4b7c9a7a51ef6e2707f2f\\\" Namespaces:<Type:\\\"mount\\\" > Namespaces:<Type:\\\"uts\\\" > Namespaces:<Type:\\\"ipc\\\" > MaskedPaths:\\\"/proc/acpi\\\" MaskedPaths:\\\"/proc/kcore\\\" MaskedPaths:\\\"/proc/keys\\\" MaskedPaths:\\\"/proc/latency_stats\\\" MaskedPaths:\\\"/proc/timer_list\\\" MaskedPaths:\\\"/proc/timer_stats\\\" MaskedPaths:\\\"/proc/sched_debug\\\" MaskedPaths:\\\"/proc/scsi\\\" MaskedPaths:\\\"/sys/firmware\\\" ReadonlyPaths:\\\"/proc/asound\\\" ReadonlyPaths:\\\"/proc/bus\\\" ReadonlyPaths:\\\"/proc/fs\\\" ReadonlyPaths:\\\"/proc/irq\\\" ReadonlyPaths:\\\"/proc/sys\\\" ReadonlyPaths:\\\"/proc/sysrq-trigger\\\" > > \" request=/grpc.AgentService/CreateContainer sandbox=7f1a1f9e64edbbe857d6883a0d013f58800cbdf832c4b7c9a7a51ef6e2707f2f source=agent\n" name=kata-proxy pid=25734 sandbox=7f1a1f9e64edbbe857d6883a0d013f58800cbdf832c4b7c9a7a51ef6e2707f2f source=agent
Aug 16 09:02:28 s8360048 kata-proxy[25734]: time="2019-08-16T09:02:28.548211122+02:00" level=info msg="time=\"2019-08-16T07:02:28.547122381Z\" level=info msg=\"Received add uevent\" debug_console=false name=kata-agent pid=1 source=agent subsystem=udevlistener uevent-action=add uevent-devname= uevent-devpath=/devices/css0/0.0.0004 uevent-seqnum=998 uevent-subsystem=css\n" name=kata-proxy pid=25734 sandbox=7f1a1f9e64edbbe857d6883a0d013f58800cbdf832c4b7c9a7a51ef6e2707f2f source=agent
Aug 16 09:02:28 s8360048 kata-proxy[25734]: time="2019-08-16T09:02:28.549550356+02:00" level=info msg="time=\"2019-08-16T07:02:28.548648012Z\" level=info msg=\"Received add uevent\" debug_console=false name=kata-agent pid=1 source=agent subsystem=udevlistener uevent-action=add uevent-devname= uevent-devpath=/devices/css0/0.0.0004/0.0.0005 uevent-seqnum=999 uevent-subsystem=ccw\n" name=kata-proxy pid=25734 sandbox=7f1a1f9e64edbbe857d6883a0d013f58800cbdf832c4b7c9a7a51ef6e2707f2f source=agent
Aug 16 09:02:28 s8360048 kata-proxy[25734]: time="2019-08-16T09:02:28.575836309+02:00" level=info msg="virtio_blk virtio4: [vda] 20971520 512-byte logical blocks (10.7 GB/10.0 GiB)\n" name=kata-proxy pid=25734 sandbox=7f1a1f9e64edbbe857d6883a0d013f58800cbdf832c4b7c9a7a51ef6e2707f2f source=agent
Aug 16 09:02:28 s8360048 kata-proxy[25734]: time="2019-08-16T09:02:28.581436036+02:00" level=info msg="time=\"2019-08-16T07:02:28.58009204Z\" level=info msg=\"Received add uevent\" debug_console=false name=kata-agent pid=1 source=agent subsystem=udevlistener uevent-action=add uevent-devname= uevent-devpath=/devices/css0/0.0.0004/0.0.0005/virtio4 uevent-seqnum=1001 uevent-subsystem=virtio\n" name=kata-proxy pid=25734 sandbox=7f1a1f9e64edbbe857d6883a0d013f58800cbdf832c4b7c9a7a51ef6e2707f2f source=agent
Aug 16 09:02:28 s8360048 kata-proxy[25734]: time="2019-08-16T09:02:28.586533736+02:00" level=info msg="time=\"2019-08-16T07:02:28.585510116Z\" level=info msg=\"Received add uevent\" debug_console=false name=kata-agent pid=1 source=agent subsystem=udevlistener uevent-action=add uevent-devname= uevent-devpath=\"/devices/virtual/bdi/254:0\" uevent-seqnum=1002 uevent-subsystem=bdi\n" name=kata-proxy pid=25734 sandbox=7f1a1f9e64edbbe857d6883a0d013f58800cbdf832c4b7c9a7a51ef6e2707f2f source=agent
Aug 16 09:02:28 s8360048 kata-proxy[25734]: time="2019-08-16T09:02:28.688725905+02:00" level=info msg="IPv6: ADDRCONF(NETDEV_CHANGE): eth0: link becomes ready\n" name=kata-proxy pid=25734 sandbox=7f1a1f9e64edbbe857d6883a0d013f58800cbdf832c4b7c9a7a51ef6e2707f2f source=agent
Aug 16 09:02:28 s8360048 kata-proxy[25734]: time="2019-08-16T09:02:28.729107592+02:00" level=info msg="time=\"2019-08-16T07:02:28.715710643Z\" level=info msg=\"Received add uevent\" debug_console=false name=kata-agent pid=1 source=agent subsystem=udevlistener uevent-action=add uevent-devname=vda uevent-devpath=/devices/css0/0.0.0004/0.0.0005/virtio4/block/vda uevent-seqnum=1004 uevent-subsystem=block\n" name=kata-proxy pid=25734 sandbox=7f1a1f9e64edbbe857d6883a0d013f58800cbdf832c4b7c9a7a51ef6e2707f2f source=agent
Aug 16 09:02:30 s8360048 kata-proxy[25734]: time="2019-08-16T09:02:30.23585353+02:00" level=info msg="time=\"2019-08-16T07:02:30.234901395Z\" level=debug msg=\"connecting vCPUs\" debug_console=false name=kata-agent pid=1 range-of-vcpus=0 sandbox=7f1a1f9e64edbbe857d6883a0d013f58800cbdf832c4b7c9a7a51ef6e2707f2f source=agent\n" name=kata-proxy pid=25734 sandbox=7f1a1f9e64edbbe857d6883a0d013f58800cbdf832c4b7c9a7a51ef6e2707f2f source=agent
Aug 16 09:02:30 s8360048 kata-proxy[25734]: time="2019-08-16T09:02:30.25649528+02:00" level=info msg="time=\"2019-08-16T07:02:30.238975657Z\" level=info msg=\"Waiting on channel for device notification\" debug_console=false devID=/devices/css0 name=kata-agent pid=1 sandbox=7f1a1f9e64edbbe857d6883a0d013f58800cbdf832c4b7c9a7a51ef6e2707f2f source=agent\n" name=kata-proxy pid=25734 sandbox=7f1a1f9e64edbbe857d6883a0d013f58800cbdf832c4b7c9a7a51ef6e2707f2f source=agent
Aug 16 09:02:33 s8360048 kata-proxy[25734]: time="2019-08-16T09:02:33.275380158+02:00" level=info msg="time=\"2019-08-16T07:02:33.258090985Z\" level=debug msg=\"request end\" debug_console=false duration=4.806354962s name=kata-agent pid=1 request=/grpc.AgentService/CreateContainer resp=\"&Empty{XXX_unrecognized:[],}\" sandbox=7f1a1f9e64edbbe857d6883a0d013f58800cbdf832c4b7c9a7a51ef6e2707f2f source=agent\n" name=kata-proxy pid=25734 sandbox=7f1a1f9e64edbbe857d6883a0d013f58800cbdf832c4b7c9a7a51ef6e2707f2f source=agent
Aug 16 09:02:33 s8360048 kata-proxy[25734]: time="2019-08-16T09:02:33.463863066+02:00" level=info msg="crw_info : CRW reports slct=0, oflw=0, chn=1, rsc=3, anc=0, erc=4, rsid=4\n" name=kata-proxy pid=25734 sandbox=7f1a1f9e64edbbe857d6883a0d013f58800cbdf832c4b7c9a7a51ef6e2707f2f source=agent
Aug 16 09:02:33 s8360048 kata-proxy[25734]: time="2019-08-16T09:02:33.503059027+02:00" level=info msg="crw_info : CRW reports slct=0, oflw=0, chn=0, rsc=3, anc=0, erc=4, rsid=0\n" name=kata-proxy pid=25734 sandbox=7f1a1f9e64edbbe857d6883a0d013f58800cbdf832c4b7c9a7a51ef6e2707f2f source=agent
Aug 16 09:02:33 s8360048 kata-proxy[25734]: time="2019-08-16T09:02:33.524356966+02:00" level=info msg="time=\"2019-08-16T07:02:33.523337492Z\" level=debug msg=\"new request\" debug_console=false name=kata-agent pid=1 req= request=/grpc.AgentService/DestroySandbox sandbox=7f1a1f9e64edbbe857d6883a0d013f58800cbdf832c4b7c9a7a51ef6e2707f2f source=agent\n" name=kata-proxy pid=25734 sandbox=7f1a1f9e64edbbe857d6883a0d013f58800cbdf832c4b7c9a7a51ef6e2707f2f source=agent
Aug 16 09:02:33 s8360048 kata-proxy[25734]: time="2019-08-16T09:02:33.537628105+02:00" level=info msg="time=\"2019-08-16T07:02:33.536733275Z\" level=debug msg=\"request end\" debug_console=false duration=12.204605ms name=kata-agent pid=1 request=/grpc.AgentService/DestroySandbox resp=\"&Empty{XXX_unrecognized:[],}\" sandbox=7f1a1f9e64edbbe857d6883a0d013f58800cbdf832c4b7c9a7a51ef6e2707f2f source=agent\n" name=kata-proxy pid=25734 sandbox=7f1a1f9e64edbbe857d6883a0d013f58800cbdf832c4b7c9a7a51ef6e2707f2f source=agent
Aug 16 09:02:33 s8360048 kata-proxy[25734]: time="2019-08-16T09:02:33.54492778+02:00" level=fatal msg="failed to handle exit signal" error="close unix @->/run/vc/vm/7f1a1f9e64edbbe857d6883a0d013f58800cbdf832c4b7c9a7a51ef6e2707f2f/kata.sock: use of closed network connection" name=kata-proxy pid=25734 sandbox=7f1a1f9e64edbbe857d6883a0d013f58800cbdf832c4b7c9a7a51ef6e2707f2f source=proxy

```

Reply:
I think you removed a little bit too much for ccw. For example https://github.com/kata-containers/agent/blob/a78e8cfda627cc350dc9d9ca9b969ebb642030c3/device.go#L60 and https://github.com/kata-containers/agent/blob/a78e8cfda627cc350dc9d9ca9b969ebb642030c3/device.go#L516 . How is the agent suppose to find the vd* device for ccw devices?

Reply:
> @lifupan it still doesn't work
> docker:
> 
> ```shell
> docker run -ti --runtime kata-runtime alpine sh
> docker: Error response from daemon: OCI runtime create failed: rpc error: code = DeadlineExceeded desc = Timeout reached after 3s waiting for device /devices/css0: unknown.
> ```
> 
> proxy:
> 
> ```shell
> Aug 16 09:02:27 s8360048 kata-proxy[25734]: time="2019-08-16T09:02:27.712174316+02:00" level=info msg="time=\"2019-08-16T07:02:27.711136358Z\" level=info msg=\"Started listening for uevents\" debug_console=false name=kata-agent pid=1 source=agent subsystem=udevlistener\n" name=kata-proxy pid=25734 sandbox=7f1a1f9e64edbbe857d6883a0d013f58800cbdf832c4b7c9a7a51ef6e2707f2f source=agent
> [...]
> 
> 
> Aug 16 09:02:28 s8360048 kata-proxy[25734]: time="2019-08-16T09:02:28.729107592+02:00" level=info msg="time=\"2019-08-16T07:02:28.715710643Z\" level=info msg=\"Received add uevent\" debug_console=false name=kata-agent pid=1 source=agent subsystem=udevlistener uevent-action=add uevent-devname=vda uevent-devpath=/devices/css0/0.0.0004/0.0.0005/virtio4/block/vda uevent-seqnum=1004 uevent-subsystem=block\n" name=kata-proxy pid=25734 
> Aug 16 09:02:30 s8360048 kata-proxy[25734]: time="2019-08-16T09:02:30.25649528+02:00" level=info msg="time=\"2019-08-16T07:02:30.238975657Z\" level=info msg=\"Waiting on channel for device notification\" debug_console=false devID=/devices/css0 name=kata-agent pid=1 sandbox=7f1a1f9e64edbbe857d6883a0d013f58800cbdf832c4b7c9a7a51ef6e2707f2f source=agent\n" name=kata-proxy pid=25734 sandbox=7f1a1f9e64edbbe857d6883a0d013f58800cbdf832c4b7c9a7a51ef6e2707f2f source=agent
> ```

Hi @alicefr, It seems you didn't used the latest patch. Can you double check is this log output by the latest agent?

Reply:
> I think you removed a little bit too much for ccw. For example
> 
> https://github.com/kata-containers/agent/blob/a78e8cfda627cc350dc9d9ca9b969ebb642030c3/device.go#L60
> 
> and
> https://github.com/kata-containers/agent/blob/a78e8cfda627cc350dc9d9ca9b969ebb642030c3/device.go#L516
> 
> . How is the agent suppose to find the vd* device for ccw devices?

This patch will get the device name from uevent instead  of got it from /sys directory. You can see in the uevent message: Aug 16 09:02:28 s8360048 kata-proxy[25734]: time="2019-08-16T09:02:28.729107592+02:00" level=info msg="time=\"2019-08-16T07:02:28.715710643Z\" level=info msg=\"Received add uevent\" debug_console=false name=kata-agent pid=1 source=agent subsystem=udevlistener uevent-action=add uevent-devname=vda uevent-devpath=/devices/css0/0.0.0004/0.0.0005/virtio4/block/vda uevent-seqnum=1004 uevent-subsystem=block\n" name=kata-proxy pid=25734 sandbox=7f1a1f9e64edbbe857d6883a0d013f58800cbdf832c4b7c9a7a51ef6e2707f2f source=agent
the "uevent-devname=vda", here is where I got it.

Reply:
got it know. I slightly modify your code
```diff
git diff
diff --git a/agent.go b/agent.go
index 67da6a7..3652e49 100644
--- a/agent.go
+++ b/agent.go
@@ -751,7 +751,7 @@ func (s *sandbox) listenToUdevEvents() {
                                                goto OUT
                                        }
                                        // blk-ccw driver case
-                                       if strings.HasSuffix(devAddress, blkCCWSuffix) {
+                                       if strings.Contains(devAddress, blkCCWSuffix) {
                                                goto OUT
                                        }
                                }
diff --git a/device.go b/device.go
index e0711e4..b10ec0a 100644
--- a/device.go
+++ b/device.go
@@ -54,7 +54,7 @@ var (
 
 // CCW variables
 var (
-       blkCCWSuffix = "virtio"
+       blkCCWSuffix = "block"
 )
 
```

and I'm getting 
```sh
docker run -ti --runtime kata-runtime alpine sh
docker: Error response from daemon: OCI runtime create failed: rpc error: code = DeadlineExceeded desc = Timeout reached after 3s waiting for device 0.0.0005/block: unknown.
```
I think the part in the agent is not triggered

Reply:
@lifupan thanks! You PR is fine on s390x. I actually found another bug, but it's not related to your PR. The rootBusPath was wrong. That's my the device was not added to the `s.pciDeviceMap[uEv.DevPath] = uEv.DevName`. My diff
```diff
git diff
diff --git a/device_s390x.go b/device_s390x.go
index 7d7d530..58138fb 100644
--- a/device_s390x.go
+++ b/device_s390x.go
@@ -8,5 +8,5 @@
 package main
 
 const (
-       rootBusPath = "/devices/pci0000:00"
+       rootBusPath = "/devices/css0"
 )

```
I'll open a PR. But you have the :heavy_check_mark: from me . And thanks so you was able to find also a bug :D

Reply:
and I really like that we get rid of looking in sys to get the device name

Reply:
> @lifupan thanks! You PR is fine on s390x. I actually found another bug, but it's not related to your PR. The rootBusPath was wrong. That's my the device was not added to the `s.pciDeviceMap[uEv.DevPath] = uEv.DevName`. My diff
> 
> ```diff
> git diff
> diff --git a/device_s390x.go b/device_s390x.go
> index 7d7d530..58138fb 100644
> --- a/device_s390x.go
> +++ b/device_s390x.go
> @@ -8,5 +8,5 @@
>  package main
>  
>  const (
> -       rootBusPath = "/devices/pci0000:00"
> +       rootBusPath = "/devices/css0"
>  )

Yeah, it's really a bug, and that's the  root cause  why this patch didn't work for you.  
 
> ```
> 
> I'll open a PR. But you have the ✔️ from me . And thanks so you was able to find also a bug :D

Reply:
Yup, I opened https://github.com/kata-containers/agent/pull/632 . So we're fine as far as both PRs get merged